addons/hr_timesheet/models/hr_timesheet.py

```diff
@@ -12,13 +12,24 @@
 class AccountAnalyticLine(models.Model):
     _inherit = 'account.analytic.line'
 
+    @api.model
+    def _get_favorite_project_id(self):
+        last_timesheet_ids = self.search([('user_id', '=', self._uid), ('project_id', '!=', False)], limit=5)
+        if len(last_timesheet_ids) == 5 and len(last_timesheet_ids.project_id) == 1:
+            return last_timesheet_ids.project_id.id
+        return False
```

Comment:
However this comment is always valid: 

No need to check if the user recorded at least 5 timesheets even if we get less we could search the default project to set.

Also I think we have to check if the `result['employee_id']` or `result['user_id']` are not defined in the `default_get` instead of directly searching the timesheets belong to the current user. 🤔 

```suggestion
        last_timesheet_ids = self.search([
            ('user_id', '=', self._uid), 
            ('project_id', '!=', False),
        ], limit=5)
        if len(last_timesheet_ids.project_id) == 1:
            return last_timesheet_ids.project_id.id
        return False
```

Reply:
> No need to check if the user recorded at least 5 timesheets even if we get less we could search the default project to set.

The purpose of this feature is to repeatedly fill the timesheet of the same project. From your suggestion, even if there is a single timesheet, the project will be set. I think behavior will change. [see](https://www.awesomescreenshot.com/video/10021735?key=961a53adf93f39e11d0bf279bfced171)

Reply:
There is no reason to not search the default project if the user has less than 5 timesheets recorded. So, it seems okay for me to improve the initial behavior.
I ask to @aju-odoo and she agrees with me. 🙂

Reply:
Done. Thanks for suggestion :slightly_smiling_face: !